io.c

```diff
@@ -11416,8 +11416,8 @@ seek_before_access(VALUE argp)
 
 /*
  *  call-seq:
- *     IO.read(command, length = nil, offset = 0, **opts) -> string or nil
- *     IO.read(path, length = nil, offset = 0, **opts)    -> string or nil
+ *     IO.read(command, length = nil, offset = 0) -> string or nil
+ *     IO.read(path, length = nil, offset = 0)    -> string or nil
```

Comment:
Oh, wow, and thanks.  I mistakenly thought that kwargs are passed as a trailing hash, and so checked using ```IO.read('t.txt', 0, 0, {})```, which fails.  My error may be repeated elsewhere; I'll check.

Reply:
That was true in Ruby 2.7 and earlier. Since 3.0 there is a separation of positional and keyword arguments.

For example:

```ruby
def foo(test:)
  puts test
end

foo({ test: "Hello world" })
```

On Ruby 2.7:

```
Hello world
```

On Ruby 3.0:

```
test.rb:1:in `foo': wrong number of arguments (given 1, expected 0; required keyword: test) (ArgumentError)
	from test.rb:5:in `<main>'
```

https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/